system/lib/wasmfs/wasmfs.cpp

```diff
@@ -6,47 +6,69 @@
 // Current Status: Work in Progress.
 // See https://github.com/emscripten-core/emscripten/issues/15041.
 
+#include "file.h"
+#include "file_table.h"
+#include "lockable.h"
 #include <emscripten/emscripten.h>
 #include <emscripten/html5.h>
+#include <errno.h>
+#include <mutex>
 #include <stdlib.h>
+#include <utility>
 #include <vector>
 #include <wasi/api.h>
 
 extern "C" {
 
-static std::vector<char> fd_write_stdstream_buffer;
+__wasi_fd_t __syscall_dup2(__wasi_fd_t oldfd, __wasi_fd_t newfd) {
+
+  // static Synchronized<FileTable> syncfileTable;
+
+  Locked<FileTable> fileTable = FileTable::get();
+
+  if (std::shared_ptr<OpenFileDescriptor>((*fileTable)[oldfd]) == nullptr) {
```

Comment:
Will this work if we keep the `bool` conversion?
```suggestion
  if (!(*fileTable)[oldfd]) {
```

Reply:
It still works, but I think we decided to keep the `bool` conversion. It's unfortunate that there is no implicit conversion by itself. Keeping the `bool` operator is the cleanest way, albeit adds an extra function.